codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -131,6 +164,18 @@ protected Result beginEncode(HttpResponse headers, String acceptEncoding) throws
                         wrapper, compressionLevel, windowBits, memLevel)));
     }
 
+    /**
+     * Returns {@code true} if the compression should be used.
+     * May be override to change default behaviour.
+     */
+    protected boolean useCompression(HttpResponse response) {
+        if (this.compressThreshold > 0) {
+            return response instanceof HttpContent &&
```

Comment:
This looks like it only works properly for a FullHttpResponse. If a HttpResponse was sent separately than a HttpContent(s), then this won't compress. Is that intended?

Reply:
This is in line with the original class.